source/agora/consensus/protocol/Nominator.d

```diff
@@ -105,36 +105,29 @@ extern(D):
 
     /***************************************************************************
 
-        Set or update the quorum configuration
-
-        The node additionally takes the list of all other validator's quorum
-        configurations and saves them for later lookup by getQSet().
+        Update our quorum configuration and store the mapping of all quorums
+        for lookup in getQSet().
 
         Params:
-            quorum = the quorum config for this node
-            other_quorums = the quorum config for all other nodes in the network
+            quorums = the mapping of all active validators' quorums
 
     ***************************************************************************/
 
-    public void setQuorumConfig (const ref QuorumConfig quorum,
-        const(QuorumConfig)[] other_quorums) nothrow @safe
+    public void setQuorumConfig (const(QuorumConfig)[PublicKey] quorums)
+        @safe nothrow
     {
         assert(!this.is_nominating);
-        () @trusted { this.known_quorums.clear(); }();
+        () @trusted { this.quorum_set.clear(); }();
 
-        // store the list of other node's quorum hashes
-        foreach (qc; other_quorums)
+        foreach (qpair; quorums.byKeyValue)  // opApply2 is not nothrow..
         {
-            auto quorum_set = buildSCPConfig(qc);
+            auto quorum_set = buildSCPConfig(qpair.value);
```

Comment:
Two suggestions about the code.
1. This name of `quroum_set` is confusing with `this.quorum_set` , so it should have another name like `temp_qset'.
2. The name of `buildSCPQuorumSet` would more representative for the operation than `buildSCPConfig`. How about changing the name?

Reply:
Good point about the naming. I was thinking the same thing, but forgot to rename it. Good suggestion about the rename too.